### PR TITLE
Change cases search sort options to a dropdown

### PR DESCRIPTION
**Pull request by stdstaff**
https://trello.com/c/AR5zgCsB/1170-update-sort-by-on-cases-page3

## Description
Update the Cases page to a newer visual design. This moves "sort by" to a dropdown on the right hand side, and changes the sort logic.

- On the initial page load of the cases page, select 'blank', but this implies "recent"
- Never show the "relevant" option
- Maintain state (keep whatever sort the user has perviously selected when applying new filters)
- If the user has 'blank' set and there's been a keyword search, the sort should still display the 'blank' option, but this should switch to mean "relevant"

## Checklist:
- [ ] Have you documented your changes in the pull request description?
- [ ] Does the change present any security considerations?
- [ ] Is any gem functionality overloaded? Eg: Devise controller methods being overloaded.
- [ ] Has acceptance criteria been tested by a peer?
- [ ] Has the CHANGELOG been updated? (If change is worth telling users about.)

### General testing (author)
- [ ] Test without JavaScript
- [ ] Test on small screen

### Accessibility testing (author)
- [ ] Works keyboard only
- [ ] Tested with one screen reader
- [ ] Zoom page to 400% - content still visible
- [ ] Disable CSS - does content make sense and appear in a logical order?

### CODE REVIEW

**Review by nasirkhanbeis — APPROVED** [review]
Tested this on the review app, Sort by dropdown works as expected. 
Also, verified the non-js version, combination with other filters and keyword search, 
sort option is retained in the session which I think is good.

File: spec/models/search_params_spec.rb [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+
+  context "when sort_by is blank" do
+    let(:search_params) { { sort_by: SearchParams::BLANK } }
+
+    describe "sort_by_option" do
+      it "returns recent by default" do
+        expect(model_instance.sort_by_option).to eq(SearchParams::RECENT)
+      end
+
+      it "returns relevant if there is a query present" do
```
Comment (KyleMacPherson) [inline spec/models/search_params_spec.rb:14]: [Minor] I would add another context for clarity here

```context "when sort_by is blank" do
        context "when there is a query present" do
        end

        context "when there is no search query present" do
        end
end```

**Review by KyleMacPherson — APPROVED** [review]

**Review by slorek — CHANGES_REQUESTED** [review]
I think we are missing a scenario in the feature spec describing the sorting behaviour when keywords are entered.

I'd also like to review the behaviour with Darren and Toby before this is introduced as I feel the changes may be confusing.

File: spec/models/search_params_spec.rb (hunk 1 above)
Comment (stdstaff) [reply spec/models/search_params_spec.rb]: :+1: https://github.com/UKGovernmentBEIS/beis-opss-psd/pull/1649/commits/5ddb635b9c966b7eec9dbf146478841ef217c679

**Comment by stdstaff** [comment]
This logic has been reviewed internally and it is felt further investigation is required into how best to present sort options to service users, and into how to implement this technically. Marking this PR "DO NOT MERGE" pending the results of these discussions.

**Review by KyleMacPherson — APPROVED** [review]

**Review by slorek — APPROVED** [review]

**Review by nasirkhanbeis — APPROVED** [review]
Tested this on the review app, looking good as expected. Nice work Nick!

**Pull request merged**